proxy/src/ctx/transport.rs

```diff
@@ -40,23 +44,30 @@ impl Server {
         local: &SocketAddr,
         remote: &SocketAddr,
         orig_dst: &Option<SocketAddr>,
+        protocol: &Protocol,
     ) -> Arc<Server> {
         let s = Server {
             proxy: Arc::clone(proxy),
             local: *local,
             remote: *remote,
             orig_dst: *orig_dst,
+            protocol: *protocol,
         };
 
         Arc::new(s)
     }
 }
 
 impl Client {
-    pub fn new(proxy: &Arc<ctx::Proxy>, remote: &SocketAddr) -> Arc<Client> {
+    pub fn new(
+        proxy: &Arc<ctx::Proxy>,
+        remote: &SocketAddr,
+        protocol: &Protocol,
```

Comment:
This could just be `Protocol`, I believe. It's just an i32. I imagine you were following the pattern of the other arguments, but in those other cases, I suppose it's because they are references to existing other things. The protocol is being created at the same time as `Client::new` is being called.

Reply:
yeah, I didn't think it needed to take a reference but wrote it that way just for consistency, I guess. looking back that doesn't make a whole lot of sense.